datafusion/core/tests/fuzz_cases/join_fuzz.rs

```diff
@@ -54,33 +55,6 @@ enum JoinTestType {
     // because if existing variants both passed that means SortMergeJoin and NestedLoopJoin also passes
     HjSmj,
 }
-#[tokio::test]
-async fn test_inner_join_1k() {
-    JoinFuzzTestCase::new(
-        make_staggered_batches(1000),
-        make_staggered_batches(1000),
-        JoinType::Inner,
-        None,
-    )
-    .run_test(&[JoinTestType::HjSmj, JoinTestType::NljHj], false)
-    .await
-}
-
-fn less_than_100_join_filter(schema1: Arc<Schema>, _schema2: Arc<Schema>) -> JoinFilter {
```

Comment:
> during NLjoin construction, join filter was completely ignored, and all NLjoin tests were running with only equijoin conditions in filter.


Just to double check my understanding, your analysis is that this test is invalid because `NestedLoopJoinExec` doesn't appy the `less_than_100_join_filter` correctly?

I reread the docs for `NestedLoopJoinExec` and it seems to imply the point is to run joins without a equality predicate 🤔 

https://docs.rs/datafusion/latest/datafusion/physical_plan/joins/struct.NestedLoopJoinExec.html

So is the difference that `less_than_100_join_filter` only referred to one join input (aka only refers to `a`) where NLJ needs filters that refer to both sides ?

Reply:
> it seems to imply the point is to run joins without a equality predicate

That's true, but in order to somehow perform fuzz testing against other join implementations, NLJoin runs (intended to run) in fuzz tests with the following filter `<on / equijoin conditions> and <optional join filter>`

> Just to double check my understanding, your analysis is that this test is invalid because NestedLoopJoinExec doesn't appy the `less_than_100_join_filter` correctly?

The test is invalid because of how NLJoin was built in these tests -- if `optional join filter` was passed as a test case argument, it was not included into NLJoin filter expression, and the test compared `join on <equijoin conditions> and <filter>` for HashJoin vs `join on <equijoin condition>` for NLJoin -- this problem is fixed by the first commit.

> So is the difference that `less_than_100_join_filter` only referred to one join input (aka only refers to a) where NLJ needs filters that refer to both sides ?

`less_than_100_join_filter` is removed because it's expression (`t.a < 100`) doesn't make much sense, since `a` produced by `make_staggered_batches` is always less than 100 -- that's why only semi/anti_filtered joins were failing (they used different filter expression -- `col_lt_col_filter`, which is not always true), and all other filtered test cases (inner/left/right/full) were passing even with missing filter for NLJoin (from p.1).

Reply:
yeah, I think I introduced the `col_lt_col_filter` when testing flaky semi join.
We probably should have used this filter for other filtered use cases.

But if the tests becoming flaky, it would be dangerous to merge the PR without ignoring tests